connection_go18.go

```diff
@@ -160,6 +160,10 @@ func (mc *mysqlConn) watchCancel(ctx context.Context) error {
 	select {
 	default:
 	case <-ctx.Done():
+		killErr := mc.kill()
+		if killErr != nil {
+			errLog.Print("failed to kill query: ", killErr)
+		}
```

Comment:
When watchCancel() is called, query is not sent yet.  Don't kill here.

Reply:
Why? The client could request too much data which will be served for really long time. Do you think we should wait for it no matter how long it takes to pass this data across the network?

Reply:
Yes, you are right: there is no reason. Error should be simply returned